### PR TITLE
Use detailed `$description` property when generating `enum` values from a `BenSampo\Enum\Enum` class

### PR DESCRIPTION
**Pull request by killtw**
- [x] Added or updated tests
- [ ] Added Docs for all relevant versions
- [x] Updated the changelog

**Changes**
Description of LaravelEnumType should using `$enum->desciption` for more detail

### CODE REVIEW

File: src/Schema/Types/LaravelEnumType.php [hunk 1]
```diff
@@ -41,8 +41,8 @@ public function __construct(string $enumClass, ?string $name = null)
                 function (Enum $enum): array {
                     return [
                         'name' => $enum->key,
-                        'value' => $enum,
-                        'description' => "$enum->value",
+                        'value' => $enum->value,
```
Comment (spawnia) [inline src/Schema/Types/LaravelEnumType.php]: The internal value should stay an instance of the enum class. That makes it work smoothly with type hints and model casts.

File: src/Schema/Types/LaravelEnumType.php [hunk 2]
```diff
@@ -41,8 +41,8 @@ public function __construct(string $enumClass, ?string $name = null)
...
+                        'description' => "$enum->description",
```
Comment (spawnia) [inline src/Schema/Types/LaravelEnumType.php]:
```suggestion
                        'description' => $enum->description,
```

File: tests/Unit/Schema/Types/LaravelEnumTypeTest.php [hunk 3]
```diff
@@ -30,6 +31,13 @@ public function testMakeEnumWithCustomName(): void
         $this->assertSame($customName, $enumType->name);
     }
 
+    public function testCustomDescription()
```
Comment (spawnia) [inline tests/Unit/Schema/Types/LaravelEnumTypeTest.php]:
```suggestion
    public function testCustomDescription(): void
```

**Comment by spawnia** [comment]
Thanks for the contribution. The test and the change are fine.

Please address the comments i left and add a line to the CHANGELOG.md

File: src/Schema/Types/LaravelEnumType.php (hunk 1 above)
Comment (killtw) [reply src/Schema/Types/LaravelEnumType.php]: But it would record instance as JSON when column type is JSON or TEXT

Comment (spawnia) [reply src/Schema/Types/LaravelEnumType.php]: Not if attribute casting is used (https://github.com/BenSampo/laravel-enum#attribute-casting).

Anyways, that has been the current behaviour for quite a while. We can't break it in a minor release. You can always use your own version of `LaravelEnumType` if you need to change that in your project.

**Pull request merged**